server/base/src/main/java/org/apache/accumulo/server/manager/state/MergeInfo.java

```diff
@@ -93,10 +100,16 @@ public boolean needsToBeChopped(KeyExtent otherExtent) {
   }
 
   public boolean overlaps(KeyExtent otherExtent) {
-    boolean result = this.extent.overlaps(otherExtent);
-    if (!result && needsToBeChopped(otherExtent))
-      return true;
-    return result;
+    return previouslyEvaluatedOverlaps.computeIfAbsent(otherExtent, b -> {
+      boolean result = this.extent.overlaps(otherExtent);
+      if (!result && needsToBeChopped(otherExtent))
+        result = true;
+      if (result)
+        LOG.debug("mergeInfo overlaps: {} true", extent);
```

Comment:
```suggestion
        LOG.debug("mergeInfo {} overlaps: {}", otherExtent, extent);
```

Reply:
This provides additional context,which may be helpful - otherwise the extent is repeated in the log as the overlap is calculated,